ruby/lib/minitest/queue/runner.rb

```diff
@@ -0,0 +1,109 @@
+require 'optparse'
+require 'minitest/queue'
+require 'ci/queue'
+require 'digest/md5'
+
+module Minitest
+  module Queue
+    class Runner
+      Error = Class.new(StandardError)
+      MissingParameter = Class.new(Error)
+
+      def self.invoke(argv)
+        new(argv).run!
+      end
+
+      attr_reader :queue_config, :options
+
+      def initialize(argv)
+        @queue_config = CI::Queue::Configuration.new
+        @argv, @options = parse(argv)
+      end
+
+      def run!
+        Minitest.queue = queue
+
+        if paths = options[:load_paths]
+          paths.split(':').reverse.each do |path|
+            $LOAD_PATH.unshift(File.expand_path(path))
+          end
+        end
+
+        @argv.each do |f|
+          require File.expand_path(f)
+        end
+
+        Minitest.queue.populate(shuffle(Minitest.loaded_tests), &:to_s) # TODO: stop serializing
```

Comment:
Is there ever a situation where we wouldn't want to shuffle here?

Reply:
Maybe, but no legitimate one. I'd rather force shuffling on everybody.